src/box/lua/net_box.lua

```diff
@@ -707,7 +758,7 @@ function remote_methods:wait_connected(timeout)
     return self:wait_state('active', timeout)
 end
 
-function remote_methods:_request(method, opts, format, stream_id, ...)
+function remote_methods:_request_no_opts_check(method, opts, format, stream_id, ...)
```

Comment:
After all we have come to a reverse of the solution I proposed at first. Make `_request` throw on critical errors like bad usage of the API and return `nil,err` on other errors (all of which are connection-related). But now we have this ugly name `no_opts_check`. Also it still throws on OOM (it can actually happen and isn't even critical AFAIK) and LuaJit errors (for instance, could be a bug where we forgot to declare a variable before usage and we will get LuaJit exception).

I am still for the solution about changing `_request` like proposed above. Yes, it will require to introduce `_request_xc()` wrapper and update some tests, but IMO it would be better than `no_opts_check` name and treating any exception like a connection-related error.

Reply:
Done